### PR TITLE
Fix at rules

### PR DESCRIPTION
**Pull request by alexander-akait**
**Description:**

fixes:
- todo
- error recovering for unknown at-rules

**BREAKING CHANGE:**

No

**Related issue (if exists):**

No

### CODE REVIEW

File: crates/swc_css_parser/src/parser/at_rule.rs [hunk 1]
```diff
@@ -1113,7 +1118,30 @@ where
                     }
                 };
 
-                // TODO validate comparison
+                let is_valid_operator = match left_comparison {
```
Comment (kdy1) [inline crates/swc_css_parser/src/parser/at_rule.rs:1121]: I think `clippy` will be angry at this.

File: crates/swc_css_parser/tests/recovery/at-rule/layer/block/output.json [hunk 2]
```diff
@@ -23,7 +23,223 @@
...
+              },
+              "token": {
+                "WhiteSpace": {
+                  "value": "   "
+                }
+              }
+            }
+          ]
+        }
+      ],
```
Comment (alexander-akait) [inline crates/swc_css_parser/tests/recovery/at-rule/layer/block/output.json:242]: We lost many tokens, because not reset state

**Review by kdy1 — APPROVED** [review]
Code changes look good, but not sure if it will pass CI.

File: crates/swc_css_parser/src/parser/at_rule.rs (hunk 1 above)
Comment (alexander-akait) [reply crates/swc_css_parser/src/parser/at_rule.rs:1121]: hm, let's check it, what is command run clippy?

Comment (kdy1) [reply crates/swc_css_parser/src/parser/at_rule.rs:1121]: You can run `cargo clippy`

Comment (alexander-akait) [reply crates/swc_css_parser/src/parser/at_rule.rs:1121]: clippy is fine :smile:

**Pull request merged**